fc/parsing/actions.py

```diff
@@ -953,7 +952,7 @@ def _add_input_variables(self):
 
             # Store initial value if given
             if variable is not None and var.initial_value is not None:
-                self.initial_values[var.rdf_term] = var.initial_value
+                self.initial_values[variable] = var.initial_value
```

Comment:
I didn't mean to resolve annotations here, since later units conversions might move the annotation. In general we only want to resolve annotations to variables at the point of need.

Reply:
Hmmm. Why is unit conversion only done at the very end of `modify_model` ?
(I can see why it's not done at this point yet, but why do it after handling defines and clamps?)

Reply:
In pycml it was done at the end, because that was how the architecture was set up - it was too complex to do it bit by bit as you went along. That might no longer be the case with cellmlmanip - and indeed we're converting time at the start. But I've not fully thought through all the details.

Regardless, it's safer only to resolve annotations to variables when needed as then you don't have to worry about other manipulations invalidating your map, and hence getting unexpected side effects.

Reply:
But... how can I set an initial value or replace an equation before units have been converted? Surely the new definition is in new units, while the variable still reports as being in the old units?

Reply:
The later pass by pycml fixed that up, putting a conversion in where the LHS & RHS of equations it had added/changed reported different units. I'm not sure if we want to do the same in weblab-fc or not. But that doesn't affect my point about the change here!

Reply:
Ok, but I have no idea what to do anymore

Reply:
Sorry, was my initial suggestion unclear? All I meant was to keep this line as `self.initial_values[var.rdf_term] = var.initial_value` and create another map from `self.initial_values` *within* the `_add_or_replace_equations` (as the first thing it does) that does variable -> value, looking up variables by ontology term at that point.

I don't think we have a test case for units converting inputs yet - when we do we'll need to figure that out. But that's independent from getting this merged so #95 works in the matching units case.